daemon/algod/api/server/v2/generated/types.go

```diff
@@ -389,6 +454,17 @@ type EvalDeltaKeyValue struct {
 	Value EvalDelta `json:"value"`
 }
 
+// KvDelta defines model for KvDelta.
+type KvDelta struct {
+	Key *[]byte `json:"key,omitempty"`
+
+	// The previous value of the KV store entry.
+	PrevValue *[]byte `json:"prev-value,omitempty"`
```

Comment:
@Eric-Warehime + reviewers Sanity checking:  Is it a by-design decision to `omitempty` for `KvDelta`s?

* Since `nil` implies specific semantics, my initial impression is:  It's _incorrect_ to `omitempty`.  Doing so, leads to arrays of different sizes.  Here's a reference:
https://github.com/algorand/go-algorand/blob/e6172bc8c415e18b4afe3ae87255e498259fdc2c/ledger/ledgercore/statedelta.go#L76-L85
* If the concern is warranted, then it may also make sense to add round-trip tests as a defense.  Here, round trip means:  internal struct -> generated struct -> JSON -> internal struct.
* I didn't consider other generated data types.  Though the concern may also apply more broadly.

Reply:
These fields are explicitly not marked as `required` in the oapi spec which is why they end up as pointers with the `omitempty` tag since a non-existent value is not really that useful to serialize.

If we want to exactly reproduce structs which have empty arrays in them we can mark them are required and provide empty arrays/slices as the value. The reason I don't think it's important to do that is that I don't know of a use case for going from `ledgercore.StateDelta` to `generated.StateDelta` and back since the SDKs generate their own types anyways.

Reply:
Resolving - @Eric-Warehime Thanks for taking a few minutes to speak verbally and clarify my confusion.  I agree omitting empty `Data` and `OldData` values still enables a downstream consumer to accurately reproduce upstream state.